cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/state/AppState.java

```diff
@@ -14,36 +14,36 @@
  * the License.
  */
 
-package io.cdap.cdap.internal.state;
+package io.cdap.cdap.internal.app.store.state;
+
+import javax.annotation.Nullable;
 
 /**
  * Represents an application state.
  */
 public class AppState {
-  private String namespace;
-  private String appName;
-  private long appId;
-  private String stateKey;
-  private byte[] stateValue;
+  private final String namespace;
+  private final String appName;
+  private final String stateKey;
+  private final byte[] stateValue;
+
+  public AppState(String namespace,
+                  String appName) {
+    this(namespace, appName, null, null);
+  }
 
   public AppState(String namespace,
                   String appName,
-                  long appId,
                   String stateKey) {
-    this.namespace = namespace;
-    this.appName = appName;
-    this.appId = appId;
-    this.stateKey = stateKey;
+    this(namespace, appName, stateKey, null);
   }
 
   public AppState(String namespace,
                   String appName,
-                  long appId,
                   String stateKey,
```

Comment:
nit: Mark the stateKey as also nullable in both constructors, getter etc since the same class is used for deleting all states.